### ISSUE TITLE
Inconsistent field behavior

### ISSUE THREAD

**Issue by jensscherbl** [symphonycms/symphony-2#2292]
I know we have already discussed this at length, but I'm getting confused again regarding the differences between an input field and a textarea field.

I remember and understand the reasoning behind wrapping unformatted textarea content in CDATA. The argument was something like _that's how potentially unsafe data is usually handled in XML_. Got it.

What I don't understand right now is, why is the input field (and maybe other core fields as well?) doing things that differently?

**Example 1**

```
Yo B*tch!

<input handle="yo-b-tch">
    Yo B*tch!
</input>
<textarea>
    <![CDATA[Yo B*tch!]]>
</textarea>
```

**Example 2**

```
Yo. <strong>This is XM & L</strong>, B*itch!

<input handle="yo-this-is-xm-and-l-b-itch">
    Yo. <strong>This is XM &amp; L</strong>, B*itch!
</input>
<textarea>
    <![CDATA[Yo. <strong>This is XM & L</strong>, B*itch!]]>
</textarea>
```

**Example 3**

```
Yo. <strong>This is not XM & L, B*itch!

<input handle="yo-this-is-not-xm-and-l-b-itch">
    Yo. &lt;strong&gt;This is not XM &amp; L, B*itch!
</input>
<textarea>
    <![CDATA[Yo. <strong>This is not XM & L, B*itch!]]>
</textarea>
```

**Comment by allen** [comment]
`HTML-encode text` should never have existed. I've disliked this option for a very long time. The decision to have this option was based on user familiarity at the time when it was implemented. HTML-encoding was a common concept to a lot of CMS/blog systems back then. We also had a hard enough time selling people on the idea of XML and XSLT, we decided not to introduce CDATA sections as another thing to worry about.

In hindsight, it was a bad choice and CDATA section is not a hard concept to grasp.

In terms of the input field and why it included HTML; this was intentional. A common practice for standard blog sites is to use input fields on article titles. However, having the HTML stripped out meant authors weren't able to format their titles properly. Also, this was before the Texbox field existed.

Hope that gives some context to the design decisions made.

Personally I'm +1 for proper use of CDATA and the obliteration of HTML Encode option.

**Comment by jensscherbl** [comment]
> Personally I'm +1 for proper use of CDATA and the obliteration of HTML Encode option.

 Makes sense. Guess we'll work on that for 3.0.0 (not sure if that's on your radar, we switched to [Semantic Versioning][1] with 2.5.0).

[1]: http://semver.org

**Comment by nitriques** [comment]
Thanks @allen !!

**Comment by jensscherbl** [comment]
Should be added to the [3.0.0][1] milestone.

[1]: https://github.com/symphonycms/symphony-2/milestones/3.0.0
